packages/catalog-model/src/entity/Entity.ts

```diff
@@ -168,6 +168,8 @@ export type EntityRelation = {
    * The entity ref of the target of this relation.
    */
   targetRef: string;
+
+  metadata?: Record<string, JsonPrimitive>;
```

Comment:
Arbitrary restriction to depth = 1. Wasn't sure of a use case to argue for either way.